src/Symfony/Component/Runtime/Runner/FrankenPhpWorkerRunner.php

```diff
@@ -23,9 +24,13 @@
  */
 class FrankenPhpWorkerRunner implements RunnerInterface
 {
+    /**
+     * @param iterable<MiddlewareInterface> $middlewares
+     */
     public function __construct(
         private HttpKernelInterface $kernel,
         private int $loopMax,
+        private iterable $middlewares = [],
```

Comment:
Looking again at the patch, there might be a simpler solution:


```suggestion
        private ?\Closure $handler = null,
```

then `$handler ??= frankenphp_handle_request(...)`

Reply:
I support Nicolas' idea, and I'd rather suggest to pass the native current handler as parameter of the custom one:
```php
$handler = // native handler ...

if ($this->handler) {
    $handler = ($this->handler)($handler);
}

$loops = 0;
do { // ...
```
Thus, we'll have the choice to replace it or wrap it.

Reply:
Thank you for your input.
Where do I define the handler in your solution?
The handler have to passed through `SymfonyRuntime`.

-----

In #62383 I have a approach to use a custom `FrankenPhpWorkerRunner`